Curtains/Curtains.Application/CurtainsService/NotifyService.cs

```diff
@@ -0,0 +1,106 @@
+using AutoMapper;
+using Curtains.Application.ConstructorObjects;
+using Curtains.Application.ConstructorObjects.Interfaces;
+using Curtains.Application.CurtainsServices.Interfaces;
+using Curtains.Application.DTO;
+using Curtains.Infrastructure.Shared.Exceptions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Curtains.Application.CurtainsService
+{
+    /// <summary>
+    /// Notifies the customer of an order
+    /// </summary>
+    class NotifyService : INotifyService
+    {
+        private const string _subject = "Вам поступил новый заказ!";
+        private readonly IMessageSender _messageSender;
+        private readonly IMessageConstructor _messageBuilder;
+        private readonly IProductImageService _productImageService;
+        private readonly IMapper _mapper;
+        private readonly IConfiguration _config;
+        private readonly ILogger _logger;
+
+        public NotifyService(IMessageSender messageSender,
+                             IMessageConstructor messageConstructor,
+                             IProductImageService productImageService,
+                             IMapper mapper,
+                             IConfiguration config,
+                             ILogger logger)
+        {
+            _messageSender = messageSender;
+            _messageBuilder = messageConstructor;
+            _productImageService = productImageService;
+            _mapper = mapper;
+            _config = config;
+            _logger = logger;
+        }
+
+        public async Task NotifyAsync(object order)
+        {
+            string message;
+            switch (order)
+            {
+                case OrderDTO orderFromCatalog:
+                    message = await GetMessageFromCatalogAsync(orderFromCatalog);
+                    break;
+                case ConstructorDTO orderFromConstructor:
+                    message = GetMessageFromConstructor(orderFromConstructor);
+                    break;
+                default:
+                    _logger.LogError("Wrong type of order");
+                    throw new WrongTypeOfOrderException(nameof(order));
+            }
+
+            await _messageSender.SendAsync(_subject, message, _config["EmailSettings:CustomerEmail"]);
+        }
+
+        /// <summary>
+        /// Sends a message to the customer based on existing products
+        /// </summary>
+        /// <param name="order">Order information and item numbers</param>
+        /// <exception cref="ArgumentNullException">Throws an error if the argument is null</exception>
+        private async Task<string> GetMessageFromCatalogAsync(OrderDTO order)
+        {
+            List<IProduct> products = new();
+            foreach (var id in order.ProductIds)
+            {
+                var productImage = await _productImageService.GetByIdAsync(id);
```

Comment:
in case of 1000 orders how many requests we will have ?

Reply:
1000?